core/store/src/trie/mod.rs

```diff
@@ -562,11 +561,11 @@ impl WrappedTrieChanges {
         store_update: &mut StoreUpdate,
     ) -> Result<(), Box<dyn std::error::Error>> {
         store_update.trie = Some(self.trie.clone());
-        for (key, raw_changes) in
+        for (key, change_with_trie_key) in
             std::mem::replace(&mut self.state_changes, Default::default()).into_iter()
         {
             assert!(
-                !raw_changes.iter().any(|RawStateChange { cause, .. }| {
+                !change_with_trie_key.changes.iter().any(|RawStateChange { cause, .. }| {
                     if let StateChangeCause::NotWritableToDisk = cause {
```

Comment:
Let's use the new [`matches!`](https://doc.rust-lang.org/stable/std/macro.matches.html)